src/main/java/seedu/address/model/person/Student.java

```diff
@@ -15,22 +15,37 @@
  */
 public class Student extends Person {
 
-    /* Each Student has 0..1 Noks */
+    /* Each Student has 0..1 Nok */
     /**
      * TODO: To prevent repeat of information, rename Person to Student such that
      *     _both_ Nok and Student inherits from Person (to share common fields)
      *     For now, Nok and Person has duplicate fields.
      */
     private Nok nok;
 
+    private ClassTiming classTiming;
+
     private final Set<Tag> tags = new HashSet<>();
 
+    /**
+     * Students have a tag as well as well as all the specifiers.
+     */
+    public Student(Name name, Phone phone, Email email, Address address, ClassTiming classTiming,
+                   Nok nok, Set<Tag> tags) {
+        super(name, phone, email, address);
+        requireAllNonNull(name, phone, email, address, classTiming, tags);
+        this.classTiming = classTiming;
+        this.nok = nok;
+        this.tags.addAll(tags);
+    }
+
     /**
      * Students have a tag as well as well as all the specifiers.
      */
     public Student(Name name, Phone phone, Email email, Address address, Nok nok, Set<Tag> tags) {
         super(name, phone, email, address);
-        requireAllNonNull(name, phone, email, address, tags);
+        requireAllNonNull(name, phone, email, address, classTiming, tags);
```

Comment:
```
super(name, phone, email, address);
requireAllNonNull(classTiming);  // this is enough
```
 super method already checks the rest are non null.
Also tags can be null...since there can be no tags right?

Reply:
This was already here when I got the code, not sure who included those inside. Tags were already inside as well.